packages/base/src/mainview/mainview.tsx

```diff
@@ -517,26 +497,30 @@ export class MainView extends React.Component<IProps, IStates> {
   };
 
   private _onSharedOptionsChanged(
-    sender: IJupyterGISDoc,
-    change: MapChange
+    sender?: IJupyterGISDoc,
+    change?: MapChange
   ): void {
     if (!this._initializedPosition) {
       const options = this._model.getOptions();
 
-      // It is important to call setZoom first, otherwise maplibre does set the center properly
-      this._Map.setZoom(options.zoom || 0);
-      this._Map.setCenter(
-        (options.longitude &&
-          options.latitude && {
-            lng: options.longitude,
-            lat: options.latitude
-          }) || [0, 0]
-      );
+      this.updateOptions(options);
 
       this._initializedPosition = true;
     }
   }
 
+  private updateOptions(options: IJGISOptions) {
```

Comment:
```suggestion
  updateOptions(options: IJGISOptions) {
```

This one should probably not be private, since it is supposed to be implemented by plugins.